### PR TITLE
support aten::sum

### PR DESCRIPTION
**Pull request by eaidova**

### CODE REVIEW

File: src/frontends/pytorch/src/op/sum.cpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+namespace op {
+
+OutputVector translate_sum(NodeContext& context) {
+    bool keep_dims = false;
+    ov::Output<ov::Node> axes;
+    Output<Node> cast;
+    FRONT_END_OP_CONVERSION_CHECK(context.get_input_size() >= 1, "Operation has no inputs.");
+    FRONT_END_OP_CONVERSION_CHECK(!context.input_is_none(0), "Input should not be None.");
+    auto data = context.get_input(0);
+    if (context.get_input_size() == 1 || context.input_is_none(1)) {
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/sum.cpp]: No need to verify input_size and is_none. `input_is_none` does this check internally.

Comment (eaidova) [reply src/frontends/pytorch/src/op/sum.cpp]: removed

File: src/frontends/pytorch/src/op/sum.cpp [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        auto start = std::make_shared<opset8::Constant>(element::i32, Shape{}, 0);
+        auto step = std::make_shared<opset8::Constant>(element::i32, Shape{}, 1);
+        auto shape = context.mark_node(std::make_shared<opset8::ShapeOf>(data, element::i32));
+        auto rank = context.mark_node(std::make_shared<opset8::ShapeOf>(shape, element::i32));
+        auto reduced_rank = context.mark_node(std::make_shared<opset8::Squeeze>(rank));
+        axes = context.mark_node(std::make_shared<opset8::Range>(start, reduced_rank, step, element::i32));
+    } else {
+        axes = context.get_input(1);
+    }
+    if (context.get_input_size() >= 2 && !context.input_is_none(2)) {
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/sum.cpp]: Please leave only `input_is_none` here too.

File: src/frontends/pytorch/src/op/sum.cpp [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+namespace ov {
+namespace frontend {
+namespace pytorch {
+namespace op {
+
+OutputVector translate_sum(NodeContext& context) {
+    bool keep_dims = false;
+    ov::Output<ov::Node> axes;
+    Output<Node> cast;
+    FRONT_END_OP_CONVERSION_CHECK(context.get_input_size() >= 1, "Operation has no inputs.");
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/sum.cpp]: The check for `input_is_none` is sufficient, no need to check input size > 0

Comment (eaidova) [reply src/frontends/pytorch/src/op/sum.cpp]: updated

File: src/frontends/pytorch/src/op/sum.cpp (hunk 2 above)
Comment (eaidova) [reply src/frontends/pytorch/src/op/sum.cpp]: updated

**Review by mvafin — APPROVED** [review]

**Pull request merged**